src/Symfony/Component/Security/Http/AccessToken/Oidc/OidcTokenHandler.php

```diff
@@ -260,4 +225,71 @@ private function decryptIfNeeded(string $accessToken): string
             return $accessToken;
         }
     }
+
+    /**
+     * Fetches the JWKS from all configured OIDC discovery endpoints.
+     *
+     * The cache entry lifetime is automatically adjusted based on the lowest TTL
+     * advertised by the providers (via "Cache-Control: max-age" or "Expires" headers).
+     */
+    private function fetchDiscoveryKeys(): array
+    {
+        $clients = $this->discoveryClients;
+        $logger = $this->logger;
+        $keys = $this->discoveryCache->get($this->oidcConfigurationCacheKey, static function (ItemInterface $item) use ($clients, $logger): array {
+            try {
+                $configResponses = [];
+                foreach ($clients as $client) {
+                    $configResponses[] = $client->request('GET', '.well-known/openid-configuration', [
+                        'user_data' => $client,
+                    ]);
+                }
+                $jwkSetResponses = [];
+                foreach ($client->stream($configResponses) as $response => $chunk) {
+                    if ($chunk->isLast()) {
+                        $jwkSetResponses[] = $response->getInfo('user_data')->request('GET', $response->toArray()['jwks_uri']);
+                    }
+                }
+                $keys = [];
+                $minTtl = null;
+                foreach ($jwkSetResponses as $response) {
+                    $currentTtl = null;
+                    $headers = array_change_key_case($response->getHeaders(), \CASE_LOWER);
+                    if (isset($headers['cache-control'][0]) && preg_match('/max-age=(\d+)/', $headers['cache-control'][0], $m)) {
+                        $currentTtl = (int) $m[1];
+                    } elseif (!empty($headers['expires'][0])) {
```

Comment:
we try to not use `empty()` in the codebase:
```suggestion
                    } elseif ($headers['expires'] ?? false) {
```

Reply:
I've removed empty().

I've opted for ``elseif (($headers['expires'][0] ?? '') !== '').``
This correctly handles all cases: non-existent headers, empty strings (['']), and the Expires: 0 case